dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/workflow/artifacts/WorkflowArtifactServiceBeanTest.java

```diff
@@ -0,0 +1,196 @@
+package edu.harvard.iq.dataverse.workflow.artifacts;
+
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowArtifact;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowArtifactRepository;
+import edu.harvard.iq.dataverse.workflow.artifacts.WorkflowArtifactServiceBean.ArtifactData;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.enterprise.inject.Vetoed;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class WorkflowArtifactServiceBeanTest {
+
+    private static final String ENCODING = "BINARY";
+    private static final String NAME = "TEST";
+
+    private static final Supplier<InputStream> TEST_DATA_SUPPLIER =
+            () -> new ByteArrayInputStream(new byte[] { 0, 1, 2, 3 });
+
+    private WorkflowArtifactServiceBean serviceBean;
+
+    private WorkflowArtifactRepository repository;
+
+    private StorageService storageService;
+
+
+    @BeforeEach
+    public void setUp() {
+        repository = new Repository();
+        serviceBean = new WorkflowArtifactServiceBean(repository);
+
+        storageService = Mockito.spy(new TestStorageService());
```

Comment:
Why you spy on the test object you have full control over? Simply add a method to access it's internal state for assertion instead.